### PR TITLE
feat(verify): multichain verification for etherscan

### LINKED ISSUE #7485: Etherscan Multichain Contract Verification
### Component

Forge

### Describe the feature you would like

Gm, we've made Etherscan's [contract verification endpoint](https://docs.etherscan.io/contract-verification/multichain-verification) accept verification requests for any [Etherscan-like explorer](https://docs.etherscan.io/contract-verification/multichain-verification) ( BscScan, PolygonScan, BaseScan etc ).

You will only need an `Etherscan API key` and append a `chainId` parameter when [submitting contract verification requests](https://docs.etherscan.io/contract-verification/multichain-verification) to Etherscan, which will verify it on the corresponding explorer.

Open to suggestions on how to best support this change in Forge Verify!

### Additional context

https://docs.etherscan.io/contract-verification/multichain-verification

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes #7485 

## Solution

Incorporate the `chain_id` parameter while building etherscan client

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm!

**Comment by yash-atreya** [comment]
@mattsse Failing cargo deny due to open-ssl being banned. This occurs after updating to `reqwest@0.12.2` and enabling the `default-tls` feature on it to use it here: 

https://github.com/yash-atreya/foundry/blob/f686351921f0d666fa726b8420aa27d312c0e0d8/crates/config/src/etherscan.rs#L323

File: crates/config/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,9 @@ Inflector = "0.11"
 number_prefix = "0.4"
 once_cell = "1"
 regex = "1"
-reqwest = { version = "0.11", default-features = false }
+reqwest = { version = "0.12", default-features = false, features = [
+    "default-tls",
```
Comment (mattsse) [inline crates/config/Cargo.toml:33]: we don't want this,
why is this enabled?

**Review by mattsse — CHANGES_REQUESTED** [review]
no openssl

File: crates/config/Cargo.toml (hunk 1 above)
Comment (mattsse) [reply crates/config/Cargo.toml:33]: this should be there with any tls feature, 
ref:

https://github.com/foundry-rs/foundry/blob/a16714ed40f733013d7a80f4f969564175c3318e/crates/forge/Cargo.toml#L107-L113

**Review by mattsse — APPROVED** [review]

**Pull request merged**